build.js

```diff
@@ -15,4 +15,7 @@ const minified = MINIFY && terser.minify(jsSourceProcessed, {
   }
 });
 
+const nodeHeader = `import { TextEncoder } from 'util';`;
+
 fs.writeFileSync('./dist/lexer.js', minified ? minified.code : jsSourceProcessed);
+fs.writeFileSync('./dist/lexer.cjs.js', `${nodeHeader}${minified ? minified.code : jsSourceProcessed}`);
```

Comment:
Ah I see, note that the dist folder is published, so this file should be outputted to `lib` instead to avoid publishing it.